Pear/Profile/ProfileSectionTableViewCell.swift

```diff
@@ -0,0 +1,114 @@
+
+        itemsCollectionView.frame = CGRect(x: 0, y: 0, width: frame.width, height: 0)
+        itemsCollectionView.dataSource = self
+        itemsCollectionView.delegate = self
+        itemsCollectionView.backgroundColor = .clear
+        itemsCollectionView.register(
+            ProfilePillCollectionViewCell.self,
+            forCellWithReuseIdentifier: ProfilePillCollectionViewCell.reuseIdentifier
+        )
+        itemsCollectionView.layoutIfNeeded()
+        contentView.addSubview(itemsCollectionView)
+
+        setupConstraints()
+    }
+
+    private func setupConstraints() {
+        titleLabel.snp.makeConstraints { make in
+            make.top.leading.trailing.equalToSuperview().offset(20)
+            make.height.equalTo(15)
+        }
+
+        itemsCollectionView.snp.makeConstraints { make in
+            make.leading.trailing.equalTo(titleLabel)
+            make.top.equalTo(titleLabel.snp.bottom).offset(8)
+            make.bottom.equalToSuperview().inset(10)
+        }
+    }
+
+    func configure(for user: User, type: ProfileSectionType) {
+        titleLabel.text = type.getTitle(for: user)
+        sectionType = type
+        switch type {
+        case .groups:
+            items = user.groups
+        case .interests:
+            items = user.interests
+        default:
+            break
+        }
+        itemsCollectionView.reloadData()
+        itemsCollectionView.sizeToFit()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+}
+
+extension ProfileSectionTableViewCell: UICollectionViewDataSource {
+}
+
+extension ProfileSectionTableViewCell: UICollectionViewDelegate {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return items.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let sectionType = sectionType,
+              let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ProfilePillCollectionViewCell.reuseIdentifier, for: indexPath) as?
+                 ProfilePillCollectionViewCell else { return UICollectionViewCell() }
+        cell.configure(with: items[indexPath.item], type: sectionType)
+        return cell
+    }
+
+}
+
+extension ProfileSectionTableViewCell: UICollectionViewDelegateFlowLayout {
+
+    func calculateNecessaryWidth(text: String) -> CGFloat {
+        let label = UILabel()
+        label.text = text
+        label.font = ._16CircularStdBook
+        label.sizeToFit()
+        return label.frame.width
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let totalHorizontalPadding: CGFloat = 52
+        return CGSize(width: calculateNecessaryWidth(text: items[indexPath.item]) + totalHorizontalPadding, height: 32)
```

Comment:
since you don't use totalHorizontalPadding again, maybe just do 
```return CGSize(width: calculateNecessaryWidth(text: items[indexPath.item]) + 52, height: 32)```